content/blog/2018/06/2018-05-14-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,205 @@
+platfoms like Bitbucket, Github, Gitlab, etc whenever a pull request is created or updated.
+
+Detect the presence of certain types of the report based on a conventional location,
+and automatically publish them. If the reports are not in a conventional location,
+users could specify the location using the YAML file.
+
+==== Benefits to the community
+* Project administrators will be able to handle builds for pull requests more easily.
+* Build specifications for pull request can be written in a more readable format.
+* Build reports will be automatically published to Github, Bitbucket, ect.
+* Build status updates will be sent to git servers automatically.
+* Users will not have to deal with pipeline code.
+
+
+This plugin will be developed on the top of the MultiBranch Pipeline plugin.
+
+.This plugin is following below steps for now
+* clone target repo
+* checkout to target branch
+* fetch the source branch
+* merge source-branch
+* call user call user script to build the repo.
+* push changes of PR to target branch
+* publish test reports
+
+Plugin will start above steps if and only if the pull request is
+mergeable, to avoid merge conflicts while merging the source branch to target
+branch. The pull request is mergebale or not can also be decided by the payload of webhook also.
+
+https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo]
+
+=== How to run the Plugin
+See https://github.com/jenkinsci/simple-pull-request-job-plugin/blob/master/README.md[How to run the demo]
+and set credentials, owner and repository on your own and you will be good to go.
+
+=== What is avalilable now
+1. User will be able to select to use Jenkinsfile.yaml for build.
+2. Git Push step
+3. harvest results and reports [and post in the PR]
+ a. junit()
+ b. findbugs()
+ c. archiveArtifacts()
+4. Basic interface to parse and get build specifications from YAML file.
+
+For now the plugin is bulding branches and Pull request both using Jnekinsfile.yaml.
+This will be fixed in next coding phase.
+
+=== Things decided
+1. To build the plugin on the top of multibranch pipeline plugin. As that plugin has implementation of
+  a. Nice interface to show diffrent branch and pull requests build seperately with use of suitable plugins like Github, Bitbucket.
+  b. Detect trusted revisions in a repository.
+  c. Pblishing of build status to the repository.
+
+2. Convert the YAML configuration to declarative pipeline.
+3. User will provide path to the script relative to the root directory of the repository
+  without extension (.sh or .bat) in the YAML file. The plugin will generate pipeline script to detect the
+  platform and call .sh or .bat script.
+
+  Example:
+    Path provided: ./scripts/hello
+    a. On UNIX machine “./scripts/hello.sh” will be called
+    b. On non-UNIX machine “./scripts/hello.bat” will be called.
+
+=== Implementations till now
```

Comment:
It would be nice to have a paragraph before it with some description. E.g. "A first prototype of the plugin is ready. It supports all features of Multi-Branch Pipeline and offers the following features:"